app/src/test/java/gov/va/vro/service/BipClaimServiceTest.java

```diff
@@ -174,6 +303,77 @@ void uploadPdf() {
     claimService.uploadPdf(payload, fetchPdfResponse);
   }
 
+  @Test
+  void testUpdateClaim() {
+    initializeBipClaims();
+    BipApiTestService bipApiService = new BipApiTestService();
+
+    ClaimProps claimProp = new ClaimProps();
+    claimProp.setSpecialIssue1(SPECIAL_ISSUES_1.get(0));
+    claimProp.setSpecialIssue2(SPECIAL_ISSUES_1.get(1));
+
+    SaveToDbService saveToDbService = Mockito.mock(SaveToDbService.class);
+    Mockito.doNothing().when(saveToDbService).updateRfdFlag(anyString(), anyBoolean());
+
+    BipClaimService claimService =
+        new BipClaimService(claimProp, bipApiService, null, saveToDbService);
+
+    for (long claimId : bipClaims.keySet()) {
+      for (MasCamelStage stage : MasCamelStage.values()) { // TODO: Is stage matters here?
+        MasProcessingObject payload = getMasPayload(claimId, stage);
+        for (MasCompletionStatus status : MasCompletionStatus.values()) {
+          BipUpdateClaimResult result = claimService.updateClaim(payload, status);
+          Set<String> specialIssueCodes = status.getSpecialIssuesToRemove(claimProps);
+          verifyUpdateClaimResult(result, payload, status, stage, specialIssueCodes);
+        }
+      }
+    }
+  }
+
+  private void verifyUpdateClaimResult(
+      BipUpdateClaimResult result,
+      MasProcessingObject payload,
+      MasCompletionStatus status,
+      MasCamelStage stage,
+      Set<String> specialIssueCodes) {
+    assertTrue(result.isSuccess());
+    long claimId = Long.parseLong(payload.getBenefitClaimId());
+    BipClaim claim = bipClaims.get(claimId);
+    ClaimStatus expectedStatus = status.getClaimStatus();
+
+    // Check claim status
+    assertEquals(expectedStatus.getDescription(), claim.getClaimLifecycleStatus());
+
+    // Check contention
+    List<ClaimContention> contentions = bipContentions.get(claimId);
+    // TODO: There might be a problem for handling multiple contentions in BipClaimService.
+    for (ClaimContention contention : contentions) {
+      // Check automation indicator.
+      if (status.isAutomationIndicator()) {
+        assertTrue(contention.isAutomationIndicator());
+      }
+
+      // Check special issue codes
+      if (!specialIssueCodes.isEmpty()) {
+        assertFalse(
+            contention.getSpecialIssueCodes().stream()
+                .anyMatch(c -> specialIssueCodes.contains(c)));
+      }
+
+      // Check contention life cycle status
+      // TODO: Should life cycle status be changed only for the contention with special codes?
```

Comment:
@au2 : I thought only the contentions with special issue codes got updated. Is the requirement to change all the contention life cycle status and automation indicator for all contentions in the claim?